opt_einsum/contract.py

```diff
@@ -621,15 +721,29 @@ def contract_expression(subscripts, *shapes, **kwargs):
     - The generated expression will work with any arrays which have
       the same rank (number of dimensions) as the original shapes, however, if
       the actual sizes are different, the expression may no longer be optimal.
+    - Constant operations will be computed upon first call with a particular
+      backend, then subsequently reused.
 
     Examples
     --------
 
-    >>> expr = contract_expression("ab,bc->ac", (3, 4), (4, 5))
-    >>> a, b = np.random.rand(3, 4), np.random.rand(4, 5)
-    >>> c = expr(a, b)
-    >>> np.allclose(c, a @ b)
-    True
+    Basic usage:
+
+        >>> expr = contract_expression("ab,bc->ac", (3, 4), (4, 5))
+        >>> a, b = np.random.rand(3, 4), np.random.rand(4, 5)
+        >>> c = expr(a, b)
+        >>> np.allclose(c, a @ b)
+        True
+
+    Supply ``a`` as a constant:
+
+        >>> expr = contract_expression("ab,bc->ac", a, (4, 5), constants=[0])
+        >>> expr
+        <ContractExpression('bc,[ab]->ac', constants=[0])>
```

Comment:
This looks like an old version where the order was changed?

Reply:
Yes good spot.